eventsources/sources/awssqs/start.go

```diff
@@ -123,6 +123,7 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 				})
 				if err != nil {
 					log.Errorw("Failed to delete message", zap.Error(err))
+					_, sqsClient, err = refreshSessionAndSqsClientIfCredentialsExpired(err, el, log)
```

Comment:
same issue here. We need to verify that the call to `refreshSessionAndSqsClientIfCredentialsExpired` was successful before reassigning the value of `sqsClient`

Reply:
Agreed. Added a nil check before reassigning sqsClient